pkgs/applications/virtualization/spike/default.nix

```
@@ -28,9 +28,24 @@ stdenv.mkDerivation rec {
 
   doCheck = true;
 
-  passthru.tests = {
-    can-run-hello-world = nixosTests.spike;
-  };
+  doInstallCheck = true;
+
+  # To test whether spike is working, we run the RISC-V hello applications using the RISC-V proxy
+  # kernel on the Spike emulator and see whether we get the expected output.
+
+  installCheckInputs =
+    let
+      riscvPkgs = import ../../../.. { crossSystem = lib.systems.examples.riscv64-embedded; };
+    in
+    [ which riscvPkgs.riscv-pk riscvPkgs.hello ];
+
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    $out/bin/spike -m64 $(which pk) $(which hello) | grep -Fq "Hello, world"
```

it seems simpler to not depend on `which` by simply writing 

```suggestion
    $out/bin/spike -m64 ${riscvPkgs.riscv-pk}/bin/pk ${riscvPkgs.hello}/bin/hello | grep -Fq "Hello, world"
```

or am i missing something?

---

That's actually how it was in the NixOS integration test. I can change it back. :)

---

Fixed.